src/types/createRouteOptions.ts

```diff
@@ -125,6 +125,10 @@ export type PropsGetter<
   TComponent extends Component = Component
 > = (route: ResolvedRoute<ToRoute<TOptions, undefined>>, context: PropsCallbackContext<TOptions>) => MaybePromise<ComponentProps<TComponent>>
 
+export type RouterViewPropsGetter<
+  TOptions extends CreateRouteOptions = CreateRouteOptions
+> = (route: ResolvedRoute<ToRoute<TOptions, undefined>>, context: PropsCallbackContext<TOptions>) => MaybePromise<RouterViewProps & Record<string, unknown>>
```

Comment:
Props generally accept additional props, and we have many tests that do not provide a component but then define random props for testing purposes. The `& Record<string, unknown>` make it so you can pass any additional props.